synapse/cortex.py

```diff
@@ -904,8 +891,38 @@ async def finidmon():
 
         await self.postNexsAnit()
 
-        if mirror is not None:
-            await self.initCoreMirror(mirror)
+        if self.mirror is not None:
+            await self._initCoreMirror(self.mirror)
+
+        # Enable leadership change awareness and fire
+        # the leadership hook once at boot
+        self.isleader = self.nexsroot.amLeader()
+        await self._initStormSvcs()
+        self.leaderchangeaware = True
+        await self.onLeaderChange(self.isleader)
+
+    async def onLeaderChange(self, leader):
+        if not self.leaderchangeaware:
+            return
```

Comment:
nop if we're not ready yet to react to changes; we may not have all our data structures ready